kyo-data/shared/src/main/scala/kyo/Result.scala

```diff
@@ -44,10 +44,16 @@ import scala.util.control.NonFatal
   */
 opaque type Result[+E, +A] >: (Success[A] | Error[E]) = Success[A] | Error[E]
 
-object Result:
+private trait LowPriorityImplicits:
+    given flatSuccess[A](using Flat[A]): Flat[Success[A]] =
+        Flat.unsafe.bypass
+
+object Result extends LowPriorityImplicits:
 
     import internal.*
 
+    type Partial[+E, +A] = Success[A] | Failure[E]
```

Comment:
An opaque type would give us more flexibility to introduce extension methods and potentially change the internal type (not sure that'd ever happen, though). 
```suggestion
    opaque type Partial[+E, +A] >: (Success[A] | Failure[E]) = Success[A] | Failure[E]
```